wrapper/src/main/java/software/amazon/jdbc/plugin/failover/FailoverConnectionPlugin.java

```diff
@@ -370,20 +370,14 @@ protected void initFailoverMode() {
       final HostSpec initialHostSpec = this.hostListProviderService.getInitialConnectionHostSpec();
       this.rdsUrlType = this.rdsHelper.identifyRdsType(initialHostSpec.getHost());
 
-      if (this.failoverMode == null) {
-        if (this.rdsUrlType.isRdsCluster()) {
-          this.failoverMode = (this.rdsUrlType == RdsUrlType.RDS_READER_CLUSTER)
-              ? FailoverMode.READER_OR_WRITER
-              : FailoverMode.STRICT_WRITER;
-        } else {
-          this.failoverMode = FailoverMode.STRICT_WRITER;
-        }
-      }
+      this.failoverMode = (this.failoverMode == null && this.rdsUrlType == RdsUrlType.RDS_READER_CLUSTER)
```

Comment:
If the user sets FAILOVER_MODE to "reader-or-writer" in their props than this will incorrectly override that setting. In that scenario this.failoverMode will be non-null on this line, so the condition will be false, so this.failoverMode will incorrectly be set to strict writer. Also, if you refactor this file I believe the failover2 plugin needs the same refactor since it has this same logic.